src/DotVVM.Framework/Resources/Scripts/metadata/coercer.ts

```diff
@@ -74,17 +74,32 @@ function tryCoerceNullable(value: any, innerType: TypeDefinition, originalValue:
 function tryCoerceEnum(value: any, type: EnumTypeMetadata): CoerceResult {
     let wasCoerced = false;
     if (typeof value === "string") {
-        if (value in type.values) {
-          return { value };
-        } else if (value !== "") {
-            // number value as string
-            const numberValue = Number(value);
-            if (!isNaN(numberValue)) {
-                value = numberValue; 
-                wasCoerced = true;
+        if (value === "") {
```

Comment:
This is quite a nasty edge case that could bite on non-flags enums. Could we support this only on flags enums that do not have a zero case?

Reply:
Actually, Newtonsoft.Json won't produce this empty string edge case when no flag is set: https://github.com/JamesNK/Newtonsoft.Json/blob/f7e7bd05d9280f17993500085202ff4ea150564a/Src/Newtonsoft.Json/Utilities/EnumUtils.cs#L203. It will probably produce value `0` or use a case like `None = 0`